en/guide/missions.md

```diff
@@ -10,13 +10,14 @@ Missions are *managed* though the [Mission](../api_reference/classdronecore_1_1_
 The [MissionItem](../api_reference/classdronecore_1_1_mission_item.md) class abstracts a small but useful subset of the mission commands supported by PX4 (and the MAVLink specification):
 
 The supported set is:
-* [MAV_CMD_NAV_WAYPOINT](http://mavlink.org/messages/common#MAV_CMD_NAV_WAYPOINT)
-* [MAV_CMD_DO_CHANGE_SPEED](http://mavlink.org/messages/common#MAV_CMD_DO_CHANGE_SPEED)
-* [MAV_CMD_DO_MOUNT_CONTROL](http://mavlink.org/messages/common#MAV_CMD_DO_MOUNT_CONTROL)
-* [MAV_CMD_IMAGE_START_CAPTURE](http://mavlink.org/messages/common#MAV_CMD_IMAGE_START_CAPTURE)
-* [MAV_CMD_IMAGE_STOP_CAPTURE](http://mavlink.org/messages/common#MAV_CMD_IMAGE_STOP_CAPTURE)
-* [MAV_CMD_VIDEO_START_CAPTURE](http://mavlink.org/messages/common#MAV_CMD_VIDEO_START_CAPTURE)
-* [MAV_CMD_VIDEO_STOP_CAPTURE](http://mavlink.org/messages/common#MAV_CMD_VIDEO_STOP_CAPTURE)
+* [MAV_CMD_NAV_WAYPOINT](https://mavlink.io/en/messages/common.html#MAV_CMD_NAV_WAYPOINT)
+* [MAV_CMD_DO_CHANGE_SPEED](https://mavlink.io/en/messages/common.html#MAV_CMD_DO_CHANGE_SPEED)
+* [MAV_CMD_DO_MOUNT_CONTROL](https://mavlink.io/en/messages/common.html#MAV_CMD_DO_MOUNT_CONTROL)
+* [MAV_CMD_IMAGE_START_CAPTURE](https://mavlink.io/en/messages/common.html#MAV_CMD_IMAGE_START_CAPTURE)
+* [MAV_CMD_IMAGE_STOP_CAPTURE](https://mavlink.io/en/messages/common.html#MAV_CMD_IMAGE_STOP_CAPTURE)
+* [MAV_CMD_VIDEO_START_CAPTURE](https://mavlink.io/en/messages/common.html#MAV_CMD_VIDEO_START_CAPTURE)
+* [MAV_CMD_VIDEO_STOP_CAPTURE](https://mavlink.io/en/messages/common.html#MAV_CMD_VIDEO_STOP_CAPTURE)
+* [MAV_CMD_NAV_LOITER_TIME](https://mavlink.io/en/messages/common.html#MAV_CMD_NAV_LOITER_TIME)

```

Comment:
Don't we need to mention first mission item in QGroundControl plan:  `MAV_CMD_NAV_TAKEOFF` ? OR Is it implicit as part of `MAV_CMD_NAV_WAYPOINT` ?

Reply:
Now this is an interesting question - and a possible bug in implementation.

If you check out [MissionItem](https://docs.dronecore.io/v/develop/en/api_reference/classdronecore_1_1_mission_item.html) there is no API to land or takeoff! (as an aside, on PX4 if the takeoff command is omitted from mission an MC vehicle will take off without it)

That is why I have this note in the guide:

> The Mission API does not (at time of writing) support "repeat", takeoff, return to land etc. DroneCore provides some omitted functionality through the Action API.

And if you look at the code we [create the missionitem](https://github.com/dronecore/DroneCore/blob/f7354e3c515dae4bd3a902d334ccb39ca42f8c10/plugins/mission/mission_impl.cpp#L972) but do nothing for the land/takeoff handling. I think that is a bug. 

Most mission created by Mission Planner will have a takeoff as their first item and land as the last item. IMO we should implement land/takeoff because they are reasonable things to want to declare, and it isn't sensible to reject import because they are missing.

@julianoes Got a better idea?

Reply:
OK. [We've considered takeoff and land](https://github.com/dronecore/DroneCore/blob/develop/plugins/mission/mission_impl.cpp#L972-L973) while importing QGC mission. But however these commands are "looked" as waypoint mission items with lat, lon, alt parameters and I remember @julianoes mentioned once that mission items are sent to firmware and firmware will simply execute them as waypoint command. @julianoes, please confirm this.

However, I feel it is better to send mission items as "takeoff", "land" commands to firmware. Reason is that
- Even after land command, vehicle stays in "Mission" & "Armed" state unless application commands for "RTL".
- Firmware won't show "Landing detected" in PX4 logs.

Reply:
> @julianoes mentioned once that mission items are sent to firmware and firmware will simply execute them as waypoint command. @julianoes, please confirm this.

I think what I meant is that you don't need to add a takeoff waypoint because the firmware will automatically insert a takeoff waypoint, so DroneCore shouldn't worry about it.

> - Even after land command, vehicle stays in "Mission" & "Armed" state unless application commands for "RTL".

That sounds like a bug. At least if auto disarm is on it should disarm after landing.

> - Firmware won't show "Landing detected" in PX4 logs.

That would also be a bug.

However, I think adding an RTL or land command at the end of the mission is just not supported by DroneCore yet.

Reply:
I tested with a QGC plan containing only takeoff and land as mission items. First I tried upload and fly using QGC. After mission accomplished, it left the vehicle in **Mission Disarmed** state. QGC Mission editor on the right showed command IDs: Takeoff, Land. Firmware shows **Landing detected** when it executes last mission item.
With DroneCore mission interface, a difference as we know is that QGC Mission editor shows both takeoff and land as Waypoint mission items. Firmware does **NOT** show **Landing detected** at the end.

Reply:
>However, I think adding an RTL or land command at the end of the mission is just not supported by DroneCore yet.

Yes

Reply:
OK, so I've never though omitting takeoff and was sensible - if only because Plane/VTOL don't auto insert them like MC does, and a waypoint at "0" altitude is not a landing - it is "very low flight" :-). 

Upshot, I agree the problem is that we don't support them yet, so let's do so: https://github.com/dronecore/DroneCore/issues/286
I haven't bothered to create issue for missing  RTL because that seems very "optional" to me.

With respect to this particular issue, my "solution" is to add a note that these are "supported for import only": https://github.com/dronecore/docs/pull/100/commits/ae773b35da304513bb5ace825156d2b7ff4c98fd - I could go into more detail on what happens in detail for import, but lets just fix this going forward. 

Merging this now and the current docs accurately reflect the API.